Disable AUTOMOUNT, .d_automount+ .d_revalidate hooks for overlayfs compatibility

  - disable snapshot dir auto mount support (.zfs/snapshot directory)
  - disable code for 'rollback of a mounted filesystem to a snapshot'
    - this breaks zfs ability to rollback over a mounted active filesystem
    - do not attempt to rollback to a mounted filesystem
  - use at your own risk

Index: zfs-linux-0.7.12/config/kernel-automount.m4
===================================================================
--- zfs-linux-0.7.12.orig/config/kernel-automount.m4
+++ zfs-linux-0.7.12/config/kernel-automount.m4
@@ -5,19 +5,4 @@ dnl # solution to handling automounts.
 dnl # which required automount support would abuse the follow_link()
 dnl # operation on directories for this purpose.
 dnl #
-AC_DEFUN([ZFS_AC_KERNEL_AUTOMOUNT], [
-	AC_MSG_CHECKING([whether dops->d_automount() exists])
-	ZFS_LINUX_TRY_COMPILE([
-		#include <linux/dcache.h>
-		struct vfsmount *d_automount(struct path *p) { return NULL; }
-		struct dentry_operations dops __attribute__ ((unused)) = {
-			.d_automount = d_automount,
-		};
-	],[
-	],[
-		AC_MSG_RESULT(yes)
-		AC_DEFINE(HAVE_AUTOMOUNT, 1, [dops->automount() exists])
-	],[
-		AC_MSG_RESULT(no)
-	])
-])
+AC_DEFUN([ZFS_AC_KERNEL_AUTOMOUNT], [AC_MSG_RESULT(no)])
Index: zfs-linux-0.7.12/module/zfs/zpl_ctldir.c
===================================================================
--- zfs-linux-0.7.12.orig/module/zfs/zpl_ctldir.c
+++ zfs-linux-0.7.12/module/zfs/zpl_ctldir.c
@@ -209,7 +209,6 @@ dentry_operations_t zpl_dops_snapdirs =
 #ifdef HAVE_AUTOMOUNT
 	.d_automount	= zpl_snapdir_automount,
 #endif /* HAVE_AUTOMOUNT */
-	.d_revalidate	= zpl_snapdir_revalidate,
 };
 
 static struct dentry *
Index: zfs-linux-0.7.12/include/linux/dcache_compat.h
===================================================================
--- zfs-linux-0.7.12.orig/include/linux/dcache_compat.h
+++ zfs-linux-0.7.12/include/linux/dcache_compat.h
@@ -76,7 +76,7 @@ d_clear_d_op(struct dentry *dentry)
 	dentry->d_op = NULL;
 	dentry->d_flags &= ~(
 	    DCACHE_OP_HASH | DCACHE_OP_COMPARE |
-	    DCACHE_OP_REVALIDATE | DCACHE_OP_DELETE);
+	    DCACHE_OP_DELETE);
 #endif /* HAVE_D_SET_D_OP */
 }
 
Index: zfs-linux-0.7.12/module/zfs/zpl_inode.c
===================================================================
--- zfs-linux-0.7.12.orig/module/zfs/zpl_inode.c
+++ zfs-linux-0.7.12/module/zfs/zpl_inode.c
@@ -823,5 +823,4 @@ const struct inode_operations zpl_specia
 };
 
 dentry_operations_t zpl_dentry_operations = {
-	.d_revalidate	= zpl_revalidate,
 };
